README.md

```diff
@@ -498,17 +498,248 @@ group members to eventually converge on a single most preferred epoch.  We have
+```
+
+### 6.2. Epoch tangle
+
+The purpose of the epoch tangle is to track progression of epochs that
+constitutes each group.
+
+For some group id `G`, where `A` is the id of the initial `group/init` message
+(in epoch 0), then define the epoch tangle for `G` as:
+1. **candate messages**:
+   - the root message with id `A` so long as it has
+     ```javascript
+     content.type = 'group/init'
+     content.tangles.epoch = { root: null, previous: null }
+     ```
+   - any messages where
+     ```javascript
+     content.type = 'group/init'
+     content.tangles.epoch = { root: A, previous: [...] }
+     ```
+2. **recipe**:
+    - a) the graph starts with the root message
+    - b) for the current tip(s) of the tangle
+        - find candidate messages which list that tip in the `previous` field   
+          of their tangle data
+        - check that the graph contains each message in `previous`
+            - if it does, add it to the graph
+            - otherwise set it aside till the graph does contain them
+        - newly added messages to the tangle graph constitue new tips
+    - c) repeat (b) till you cannot grow the tangle further
+
+```mermaid
+flowchart RL
+
+subgraph epoch_0[Epoch 0]
+  0_init[init]
+end
+
+subgraph epoch_1[Epoch 1]
+  1_init[init]
+end
+
+%% epoch tangle
+1_init --> 0_init
+linkStyle 0 stroke:#EF476F,stroke-width:2;
+
+
+classDef default stroke:#ccc,fill:#eee,color:#333;
+classDef cluster fill:#fff,stroke:#000,color:#333;
+```
+
+### 6.3. Group tangle
+
+The purpose of the group tangle is to clearly identify all messages which are
+part of a particular group, and provide partial causal ordering.
+
+For some group id `G`, where `A` is the id of the initial `group/init` message
+(in epoch 0), then define the group tangle for `G` as:
+1. **candate messages**:
+   - the root message with id `A` so long as it has
+     ```javascript
+     content.type = 'group/init'
+     content.tangles.group = { root: null, previous: null }
+     ```
+   - any messages where
+     ```javascript
+     content.tangles.group = { root: A, previous: [...] }
+     ```
+2. **recipe**:
+    - a) the graph starts with the root message
+    - b) for the current tip(s) of the tangle
+        - find candidate messages which list that tip in the `previous` field   
+          of their tangle data
+        - check that the graph contains each message in `previous`
+            - if it does, add it to the graph
+            - otherwise set it aside till the graph does contain them
+        - newly added messages to the tangle graph constitue new tips
+    - c) repeat (b) till you cannot grow the tangle further
+
+<!--
+```mermaid
+flowchart RL
+
+M -.-> X & Y -.-> B -.-> A
+
+linkStyle 0,1,2,3,4 stroke:#FFD166,stroke-width:4;
+
+classDef default stroke:none,fill:#ccf,color:#333;
+classDef cluster fill:#fff,stroke:#000,color:#333;
+````
+-->
 
-### 6.1. Normative References
+
+### 6.4. Using all the tangles together
+
```

Comment:
```suggestion

In this section we provide an example that illustrates the presence of all three group-related tangles.
Suppose peer Z creates a group and performs the following actions, in this order:

1. Adds peer A to the group at epoch 0
2. Adds peers B and C to the group at epoch 0
3. Excludes peer C, thus creating epoch 1
4. Re-adds peers A and B to the group, now at epoch 1

Then the following diagram illustrates Z's feeds (additions feed, epoch 0 feed, epoch 1 feed) and its
messages:
```

Reply:
why do you want to remove this helpful description of what the diagram shows?

Reply:
Remove?? My commit suggestion is *adding* the description.

Reply:
weird, my brain was poo apparently